packages/data/src/factory.js

```diff
@@ -58,15 +51,6 @@ export function createRegistrySelector( registrySelector ) {
 	 */
 	selector.isRegistrySelector = true;
 
-	/**
-	 * Registry on which to call `select`. The `defaultRegistry` value is overwritten with
-	 * the actual registry when the selector is registered with a store. Therefore, it's never
-	 * used and is assigned here only to satisfy the type constraint of the `registry` field.
-	 *
-	 * @type {WPDataRegistry}
-	 */
-	selector.registry = defaultRegistry;
```

Comment:
I described the motivation for this in the commit's commit message:

> We intend to register some controls created by `createRegistryControl` as built-ins with every store. In order to do that, we need to break a dependency cycle where registry creation depends on store registration which depends on creating controls which depends on default registry.
>
> The solution is to remove the `defaultRegistry` binding, which is there only to satisfy a typechecker anyway, and doesn't have any runtime impact.

Another solution would be to break up the `factory.js` module and put `createRegistrySelector` and `createRegistryControl` into separate modules. `createRegistrySelector` has the offending `defaultRegistry` dependency, while `createRegistryControl` is needed to create a store and therefore introduces a cyclic dependency.

However, to make the API really type-safe, we'd need to rethink how we create the selectors and introduce types like `UnregisteredRegistrySelector` and `RegisteredRegistrySelector`. Right now, the usage is like:
```js
const selector = createRegistrySelector( ( select ) => ( state ) => { select( 'core' ) /* ... */ } );

// at this moment the selector is in a weird state where it's either unbound or bound to `defaultRegistry`
// it's not entirely certain whether the following call should crash or not and what `value` is supposed to be
const value = selector( state ); 

// only after this registration is the `selector` registered and has 100% defined behavior
customRegistry.registerStore( 'mystore', { reducer, selectors:  { selector } } );

// this second call can have an entirely different result from the first one, although the `selector`
// value is still the same and has the same type. The `selector` properties have been mutated during
// `registerStore`.
const value = selector( state );
```
I.e., the code is not safe and I don't immediately see how introducing types could prevent the unsafe usages.

Reply:
I'm not sure i understand everything here but I agree that depending on the default registry should be avoided anyway. The default registry only exists because it was there first. If we'd start over, we'd probably avoid it entirely.

Reply:
> I'm not sure i understand everything here but I agree that depending on the default registry should be avoided anyway.

The tl;dr is that the default registry reference started introducing dependency cycles 🙂